bootstrap/bootstrap.go

```diff
@@ -704,6 +704,13 @@ func (b *Bootstrap) CheckoutPhase() error {
 			err := b.defaultCheckoutPhase()
 			if err != nil {
 				b.shell.Warningf("Checkout failed! %s (%s)", err, s)
+
+				// Checkout can fail because of corrupted files in the checkout
+				// which can leave the agent in a state where it keeps failing
+				// This removes the checkout dir, which means the next checkout
+				// will be a lot slower (clone vs fetch), but hopefully will
+				// allow the agent to self-heal
+				_ = b.removeCheckoutDir()
```

Comment:
This LGTM.

I'd love to see the agent also maintain a `bare` clone elsewhere as a local mirror, to optimise away the network trip?

Reply:
Me too, but the scope of that is a lot bigger I think.

Reply:
This would be *killer*.

Reply:
💯 totally agree. Would happily accept a PR if you'd like to see it sooner @petemounce @jgavris :) - but it's something we absolutely want to do in the future!